src/main/update/check.ts

```diff
@@ -0,0 +1,112 @@
+import { IApv, ISimpleApv } from "src/interfaces/apv";
+import { IDownloadUrls, getDownloadUrls } from "../../utils/url";
+import Headless from "../headless/headless";
+
+export class GetPeersApvFailedError extends Error {}
+
+export interface IUpdateContext {
+  newApv: ISimpleApv;
+  oldApv: ISimpleApv;
+  urls: IDownloadUrls;
+}
+
+export async function checkUpdateRequired(
```

Comment:
```suggestion
export async function checkForUpdate(
```

`checkUpdateRequired` sounds like we're declaring we need to check for an update. The function name should represent what it does as a verb.

Reply:
Done